### PR TITLE
Add checks after request leadership

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

This diff checks if everything is properly updated before the follower coordinator becomes LEADER on user action. It is still possible for user to do action while failover is happening but now we have a way to detect that everything is properly set up.

[master < Task] PR
- [ ] Provide the full content or a guide for the final git message
    - Add setup check on leadership change

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug/feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - Add check to confirm correct setup after requesting leadership 
- [x] Link the documentation PR here
    - No docs only changelog
- [x] Tag someone from docs team in the comments

### CODE REVIEW

**Comment by antoniofilipovic** [comment]
@kgolubic tagging you here, but this is for 2.17 release

File: src/coordination/coordinator_instance.cpp [hunk 1]
```diff
@@ -354,6 +358,9 @@ void CoordinatorInstance::ForceResetCluster() {
   std::ranges::for_each(repl_instances_, [](auto &instance) { instance.StartFrequentCheck(); });
 
   MG_ASSERT(!raft_state_.IsLockOpened(), "After force reset we need to be in healthy state.");
+  auto leader_lock = std::unique_lock{leader_mutex_};
+  leader_set_up_.store(true);
+  become_leader_.notify_all();
```
Comment (as51340) [inline src/coordination/coordinator_instance.cpp:363]: Why is this done in`ForceReset`? Also, what is the meaning of `leader_set_up_`, why it is saved in coordinator_instance?

File: src/coordination/coordinator_instance.cpp [hunk 2]
```diff
@@ -539,9 +548,39 @@ auto CoordinatorInstance::SetReplicationInstanceToMain(std::string_view instance
...
+  auto maybe_stop = utils::ResettableCounter<5>();
+  do {
+    spdlog::trace("Waiting for coordinator to become leader.");
+    if (!become_leader_.wait_for(lock, waiting_period, [this]() { return leader_set_up_.load(); })) {
+      waiting_period *= expontential_backoff;
+    }
+  } while (!maybe_stop() && !leader_set_up_);
+
+  return leader_set_up_;
+}
```
Comment (as51340) [inline src/coordination/coordinator_instance.cpp:570]: I understand this, we are waiting for callback to be executed to make sure we are the leader, right?

File: src/coordination/coordinator_instance.cpp (hunk 1 above)
Comment (as51340) [reply src/coordination/coordinator_instance.cpp:363]: Doesn't `ForceReset` only change status of replication instances?

**Review by as51340 — CHANGES_REQUESTED** [review]

File: src/coordination/coordinator_instance.cpp (hunk 1 above)
Comment (antoniofilipovic) [reply src/coordination/coordinator_instance.cpp:363]: Change as talked on call

**Comment by antoniofilipovic** [comment]
This PR will get closed as we discussed and we won't introduce any checks on request leadership rather completely remove it

**Pull request closed without merging**